R/financial_functions.R

```diff
@@ -16,7 +16,37 @@
 #'   under baseline and late and sudden, market prices/costs, company net profit
 #'   margins, the proximity to target in the production forecast period and an
 #'   indication of the direction of the technology.
-calculate_net_profits <- function(data) {
+#' @param shock_year A numeric vector of length one that indicates in which year
+#'   the policy shock strikes in a given scenario.
+#' @param end_year Numeric, holding end year of analysis.
+#' @param carbon_data NGFS carbon prices.
+calculate_net_profits <- function(data,
+                                  shock_year,
+                                  end_year,
+                                  carbon_data) {
+
+  carbon_data <- carbon_data %>% dplyr::filter(.data$model == "MESSAGEix-GLOBIOM 1.0") %>% dplyr::select(-c(scenario_geography))
```

Comment:
shouldnt this happen in process_data? 
Also why is geo in the data when it is deselected? 
Also please do not hard code in the code (e.g. MESSAGEix-GLOBIOM 1.0), if it is not evitable please add it as a lookup.R, compare e.g. end_year_lookup  :)

Reply:
well the thing is i now wanna integrate carbon tax as a user defined input, i.e. they will be able to select the model they want to use so i will get rid of the hardcoded value now in the next step. is that ok and sensible? geography is in there bc i intend to have a better dataset at some point that contains different carbon prices for different regions